src/main/java/uk/gov/hmcts/reform/hmc/service/common/OverrideAuditService.java

```diff
@@ -0,0 +1,161 @@
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OverrideAuditService {
+
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final String PARAM_ATTRIBUTE =
+        "org.springframework.web.servlet.HandlerMapping.uriTemplateVariables";
+
+    private final HearingStatusAuditRepository hearingStatusAuditRepository;
+    private final LinkedHearingStatusAuditRepository linkedHearingStatusAuditRepository;
+    private final UrlManager roleAssignmentUrlManager;
+    private final UrlManager dataStoreUrlManager;
+
+    // To overcome circular dependency issue
+    @Lazy @Autowired @Setter
+    private SecurityUtils securityUtils;
+
+    public void logOverrideAudit(HttpServletRequest request) {
+        String roleAssignmentUrl = request.getHeader(roleAssignmentUrlManager.getUrlHeaderName());
+        String dataStoreUrl = request.getHeader(dataStoreUrlManager.getUrlHeaderName());
+
+        if (!isLogRequired(roleAssignmentUrl, dataStoreUrl)) {
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode root = mapper.createObjectNode();
+
+        String path = request.getRequestURI();
+        root.put("path", path);
+
+        if (roleAssignmentUrl != null && !roleAssignmentUrlManager.getHost().equals(roleAssignmentUrl)) {
+            root.put("role-assignment-url", roleAssignmentUrl);
+        }
+
+        if (dataStoreUrl != null && !dataStoreUrlManager.getHost().equals(dataStoreUrl)) {
+            root.put("data-store-url", dataStoreUrl);
+        }
+
+        root.put("hmctsServiceName", getServiceName(request.getHeader(SERVICE_AUTHORIZATION)));
+
+        if (isRequestWithBody(request)) {
+            root.put("requestBody", getRequestBody(request));
+        }
+
+        root.put("requestTimestamp", LocalDateTime.now(ZoneId.of("UTC")).toString());
+        root.put("user-id", securityUtils.getUserId());
+
+        if (path.startsWith("/linkedHearingGroup")) {
+            String groupId = getAttributeId(request, "id");
+            saveLinkedHearingStatusAudit(groupId, root);
+        } else {
+            String hearingId = getAttributeId(request, "id");
+            saveHearingStatusAudit(hearingId, root);
+        }
+    }
+
+    private void saveLinkedHearingStatusAudit(String groupId, ObjectNode root) {
+        LinkedHearingStatusAuditEntity auditEntity = new LinkedHearingStatusAuditEntity();
+        auditEntity.setHmctsServiceId("n/a");
+        auditEntity.setLinkedGroupId(groupId);
+        auditEntity.setLinkedGroupVersion("n/a");
+        auditEntity.setLinkedHearingEvent("n/a");
+        auditEntity.setOtherInfo(root);
+        linkedHearingStatusAuditRepository.save(auditEntity);
+    }
+
+    private void saveHearingStatusAudit(String hearingId, ObjectNode root) {
+        HearingStatusAuditEntity auditEntity = new HearingStatusAuditEntity();
+        auditEntity.setHearingId(hearingId);
+        auditEntity.setHmctsServiceId("n/a");
```

Comment:
service id we can obtain from the create and update payloads

Reply:
This would require quite a lot of checking on what kind of incoming request is. Given that it is fairly light traffic and that information is logged as per standard way that already exists, I felt that it shouldn't be hard to match those two audit logs. In addition, I do save the request body on otherInfo field as "requestBody", so in theory it would be possible to extract information from DB even without that second log entry. (by second I mean the one that should insert audit entries on save/update requests)

Reply:
Happy with the above not being applied